dbcon/mysql/ha_calpont_execplan.cpp

```diff
@@ -5150,6 +5190,16 @@ void gp_walk(const Item* item, void* arg)
 
                 case STRING_RESULT:
                 {
+                    // Special handling for 0xHHHH literals
+                    const Type_handler *tph = item->type_handler();
+                    if (dynamic_cast<const Type_handler_hex_hybrid*>(tph) != NULL)
+                    {
+                        Item_hex_hybrid *hip = (Item_hex_hybrid*) item;
+                        gwip->rcWorkStack.push(new ConstantColumn((int64_t)hip->val_int(), ConstantColumn::NUM));
+                        (dynamic_cast<ConstantColumn*>(gwip->rcWorkStack.top()))->timeZone(gwip->thd->variables.time_zone->get_name()->ptr());
```

Comment:
Heavy to read. Kindly separate operations to make it readable.

Reply:
Done.